### PR TITLE
fix: clippy --all

### PR DESCRIPTION
**Pull request by willemneal**
### What

Ensure that clippy used across repo. Also use an action and split CLI integration tests from normal ones. 

### Why

Consistency and make git hooks.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: cmd/crates/soroban-test/tests/fixtures/test-wasms/udt/src/lib.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub struct Contract;
 
 #[contractimpl]
 impl Contract {
+    #[allow(clippy::unnecessary_fold)]
```
Comment (leighmcculloch) [inline cmd/crates/soroban-test/tests/fixtures/test-wasms/udt/src/lib.rs:37]: Still needed if we switch to sum?

**Review by leighmcculloch — APPROVED** [review]

File: cmd/crates/soroban-test/tests/fixtures/test-wasms/udt/src/lib.rs (hunk 1 above)
Comment (willemneal) [reply cmd/crates/soroban-test/tests/fixtures/test-wasms/udt/src/lib.rs:37]: I thought it was a nice example for readers that they could do it both ways.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ jobs:
     - run: rustup target add wasm32-unknown-unknown
     - if: matrix.target == 'aarch64-unknown-linux-gnu'
       run: sudo apt-get update && sudo apt-get -y install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
-    - run: cargo clippy --all-targets --target ${{ matrix.target }}
+    - run: cargo clippy --all --all-targets --target ${{ matrix.target }}
```
Comment (willemneal) [inline .github/workflows/rust.yml:65]:
```suggestion
    - run: cargo clippy --all-targets --target ${{ matrix.target }}
```

**Comment by willemneal** [comment]
@2opremio, I had to update the budget to pass the tests, but the fee dropped around 50%; this seems extreme, is it correct?

**Pull request merged**